File: frontend/src/types/chrono.ts
export interface DocumentChrono {
  id: string;
  nom_fichier_original: string;
  statut: string;
  categorie: string | null;
  annee: number | null;
  mois: number | null;
  entreprise_id: string | null;
  entreprise_nom: string | null;
  created_at: string;

  ecriture_id: string | null;
  numero_piece: string | null;
  date_piece: string | null;
  tiers: string | null;
  montant_ht: string | null;
  taux_tva: string | null;
  montant_tva: string | null;
  montant_ttc: string | null;
  statut_validation: string | null;
  anomalie_detectee: boolean;
  anomalie_details: string | null;
  saisie_topaze: boolean;
}

File: backend/app/schemas/chrono.py
"""Schéma de la vue Chronos : document + écriture associée."""

import uuid
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict


class DocumentChronoOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    nom_fichier_original: str
    statut: str
    categorie: str | None = None
    annee: int | None = None
    mois: int | None = None
    entreprise_id: uuid.UUID | None = None
    entreprise_nom: str | None = None
    created_at: datetime

    ecriture_id: uuid.UUID | None = None
    numero_piece: str | None = None
    date_piece: date | None = None
    tiers: str | None = None
    montant_ht: Decimal | None = None
    taux_tva: str | None = None
    montant_tva: Decimal | None = None
    montant_ttc: Decimal | None = None
    statut_validation: str | None = None
    anomalie_detectee: bool = False
    anomalie_details: str | None = None

    # Désormais porté par Document pour fonctionner aussi avec Banque.
    saisie_topaze: bool = False
